File: libusbcamera/src/main/java/com/rain/uvc/provider/OverallContextProvider.java
package com.rain.uvc.provider;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

/**
 * @author yuan
 * @createTime: 2024/8/16
 * @des 全局context实例初始化的contextProvider，一般用于lib初始化，解决lib库内不存在context实例的问题
 */
public class OverallContextProvider extends ContentProvider {
    @Override
    public boolean onCreate() {
        OverallContext.init(getContext());
        return true;
    }

    @Override
    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
        return null;
    }

    @Override
    public String getType(Uri uri) {
        return "";
    }

    @Override
    public Uri insert(Uri uri, ContentValues values) {
        return null;
    }

    @Override
    public int delete(Uri uri, String selection, String[] selectionArgs) {
        return 0;
    }

    @Override
    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
        return 0;
    }
}

File: app/src/main/java/com/rain/uvc/demo/provider/OverallContextProvider.kt
package com.rain.uvc.demo.provider

import android.app.Application
import android.content.ContentProvider
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.net.Uri

/**
 *  全局context实例对象
 *
 *  Describe：context初始化类
 *  History：修改记录：【作者】：【时间】：【修改内容】
 */
object OverallContext {
	@JvmStatic
	lateinit var baseContext: Application
	
	@JvmStatic
	internal fun init(context: Context?) {
		baseContext = context?.applicationContext as Application
	}
}

internal class OverallContextProvider : ContentProvider() {
	override fun onCreate(): Boolean {
		OverallContext.init(context)
		return true
	}
	
	override fun query(uri: Uri, projection: Array<out String>?, selection: String?, selectionArgs: Array<out String>?, sortOrder: String?): Cursor? = null
	
	override fun getType(uri: Uri): String? = null
	
	override fun insert(uri: Uri, values: ContentValues?): Uri? = null
	
	override fun delete(uri: Uri, selection: String?, selectionArgs: Array<out String>?) = 0
	
	override fun update(uri: Uri, values: ContentValues?, selection: String?, selectionArgs: Array<out String>?) = 0
}